csup/context.go

```diff
@@ -0,0 +1,70 @@
+package csup
+
+import (
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/brimdata/super"
+	"github.com/brimdata/super/sup"
+	"github.com/brimdata/super/zio/bsupio"
+)
+
+type Context struct {
+	mu     sync.Mutex
+	local  *super.Context // holds the types for the Metadata values
```

Comment:
Maybe call this sctx for consistency?

Reply:
This raises an interesting point. I find it helpful to know it's different than the shared query context (i.e., created locally vs passed in) so that you need to do translations.